Add new settings introduced in ST 4114 to the preferences schema

Ultra-Instinct-05 opened the pull request:
This PR adds the 3 new settings introduced in ST 4114 that allows a user to configure the number of matches/file sizes when using certain `Find` options. More details can be seen in the commits.

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1262,21 @@
+          "default": 8192
+        },
+        "find_highlight_matches_max_size": {
+          "markdownDescription": "The maximum file size (in bytes) for which the find option **Highlight matches** will highlight matches. Use 0 to remove this limit",
```
```suggestion
          "markdownDescription": "The maximum file size (in bytes) for which the find option **Highlight matches** will highlight matches (affects non-regexp searches). Use `0` to remove this limit.",
```

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1262,21 @@
+          "default": 16777216
+        },
+        "find_regex_highlight_matches_max_size": {
+          "markdownDescription": "The maximum file size (in bytes) for which the find option **Regular expression** will highlight matches. Use 0 to remove this limit",
```
```suggestion
          "markdownDescription": "The maximum file size (in bytes) for which the find option **Highlight matches** will highlight matches (non-regexp searches). Use `0` to remove this limit",
```

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1262,21 @@
           "default": 16777216
+        },
+        "find_scroll_highlights_limit": {
+          "markdownDescription": "The maximum number of find results to show in the scroll bar. If this number is exceeded no results will be shown. Use 0 to remove this limit.",
```
```suggestion
          "markdownDescription": "The maximum number of find results to show in the scroll bar. If this number is exceeded no results will be shown. Use `0` to remove this limit.",
```

rchl commented:
Will squash so you don't have to care about fixing the history up.

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1262,21 @@
+          "default": 16777216
+        },
+        "find_regex_highlight_matches_max_size": {
+          "markdownDescription": "The maximum file size (in bytes) for which the find option **Regular expression** will highlight matches (affects regexp searches). Use `0` to remove this limit.",
```
My suggestion included more changes because I thought it would be more syntactically correct. IMO both `find_highlight_matches_max_size` and `find_regex_highlight_matches_max_size` should be describing the functionality in terms of the `Highlight matches` option, not `Regular expression`,  since they affects the matches highlighting.

Ultra-Instinct-05 replied:
Agreed. Sorry about this, looks like I did not read the previous review carefully.

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1262,21 @@
           "type": "integer",
           "default": 16777216
+        },
+        "find_scroll_highlights_limit": {
+          "markdownDescription": "The maximum number of find results to show in the scroll bar. If this number is exceeded no results will be shown. Use `0` to remove this limit.",
+          "type": "integer",
+          "default": 8192
+        },
+        "find_highlight_matches_max_size": {
+          "markdownDescription": "The maximum file size (in bytes) for which the find option **Highlight matches** will highlight matches (affects non-regexp searches). Use `0` to remove this limit.",
+          "type": "integer",
+          "default": 16777216
+        },
+        "find_regex_highlight_matches_max_size": {
```
I believe all of those should be within the `EditorSettings` object as those appear to be supported in syntax-specific or project settings.

Ultra-Instinct-05 replied:
I don't think I understand ? What `EditorSettings` object ?

rchl replied:
Those settings should be inserted within this `properties` object: https://github.com/sublimelsp/LSP-json/blob/abfb435901e8cf4c2b1ec988842482f05287bdf4/schemas/preferences.sublime-settings.json#L9-L10

Ultra-Instinct-05 replied:
Now that I look at this `preferences.sublime-settings.json` more closely, it seems to be divided into `EditorSettings` and `allOf`. What's the difference ?

rchl replied:
The preferences within `EditorSettings` are also supported in syntax-specific and project settings while all other are not.

rchl replied:
Since I was pinged, I'll point out that this is still not addressed.

Ultra-Instinct-05 replied:
Yes. I am aware that I have not addressed this. I had a busy last week. I will get this addressed this weekend. There appears to be a couple more settings added in 4114 that also need to go along with this PR that I missed adding.

Ultra-Instinct-05 replied:
I have addressed this.

Ultra-Instinct-05 commented:
Two more settings were added to 4114 (diffing between 4112 & 4114), which are now added. I have added them to `allOf` since they seem to be used only in preferences and don't seem to be something used in syntax or project settings.

Also apologies I took way longer to work on review comments than it should have :(

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -855,6 +855,21 @@
           "default": true
+        },
+        "find_scroll_highlights_limit": {
+          "markdownDescription": "The maximum number of find results to show in the scroll bar. If this number is exceeded no results will be shown. Use `0` to remove this limit.",
```
I think there is a missing comma or `then` here.

```suggestion
          "markdownDescription": "The maximum number of find results to show in the scroll bar. If this number is exceeded then no results will be shown. Use `0` to remove this limit.",
```

(I know this is probably from ST itself but those are not always entirely correct either. :))

Ultra-Instinct-05 replied:
Taken care of all the points.

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1277,16 @@
           "default": 16777216
+        },
+        "focus_on_file_drop": {
+          "markdownDescription": "Whether Sublime Text should take focus when a file is dragged & dropped into it. On **Windows**, this is overriden to true in the platform specific settings.",
```
```suggestion
          "markdownDescription": "Whether Sublime Text should take focus when a file is dragged & dropped into it. On **Windows**, this is overridden to `true` in the platform specific settings.",
```

rchl commented on schemas/preferences.sublime-settings.json:
```diff
@@ -1262,6 +1277,16 @@
+          "default": false
+        },
+        "remember_layout": {
+          "markdownDescription": "Whether Sublime Text should start with the same layout it was closed with. Only applies when **hot_exit** setting is disabled.",
```
```suggestion
          "markdownDescription": "Whether Sublime Text should start with the same layout it was closed with. Only applies when `hot_exit` setting is disabled.",
```

rchl reviewed (commented):
Minor changes requested.
I wouldn't bother you if I could push to your PR but that's not possible. :)

rchl commented:
Thanks!
